src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,375 @@
+package duke;
+
+import exception.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class TaskManager {
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_DELETE = "delete";
+
+    private static final String SEPARATOR_SPACE = " ";
+    private static final String SEPARATOR_BY = " /by ";
+    private static final String SEPARATOR_AT = " /at ";
+    private static final String SEPARATOR_VERTICAL_LINE = " \\| ";
+    private static final String SEPARATOR_WORD_FILE = " | ";
+
+    private static final int LENGTH_TODO = 4;
+    private static final int LENGTH_DEADLINE = 8;
+    private static final int LENGTH_EVENT = 5;
+
+    private static final String TASK_IS_DONE = "1";
+    private static final String TASK_IS_NOT_DONE = "0";
+
+    private static final String INITIAL_TODO = "T";
+    private static final String INITIAL_DEADLINE = "D";
+    private static final String INITIAL_EVENT = "E";
+
+    private static final String DATA_FILEPATH = "duke.txt";
+
+    private static final boolean FOR_LOADING = true;
+    private static final boolean NOT_FOR_LOADING = false;
+
+    private static final String DIVIDER = "    ____________________________________________________________";
+
+    private List<Task> tasks;
+
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+    }
+
+    private int getTasksSize() {
+        return tasks.size();
+    }
+
+    public void loadData() {
+        try {
+            loadSavedData();
+        } catch (FileNotFoundException e) {
+            File file = new File(DATA_FILEPATH);
+            try {
+                file.createNewFile();
+            } catch (IOException ex) {
+                System.out.println("     Unable to create new file");
+            }
+
+        }
+    }
+
+    public void loadSavedData() throws FileNotFoundException {
+        File f = new File(DATA_FILEPATH);
+        Scanner s = new Scanner(f);
+
+        while(s.hasNext()) {
+            loadSavedInputToTasks(s.nextLine());
+        }
+    }
+
+    private void loadSavedInputToTasks(String input) {
+        String[] inputs = input.split(SEPARATOR_VERTICAL_LINE);
+        String taskType = inputs[0];
+        String isTaskDone = inputs[1];
+        String taskDescription = inputs[2];
+
+        if (taskType.equals(INITIAL_TODO)) {
+            addTodoTaskWithException(COMMAND_TODO + " " + taskDescription, FOR_LOADING);
```

Comment:
The parameter to be passed to the `addTodoTaskWithException()` could be assigned a variable name first. This way makes it difficult to read.